### PR TITLE
Splitting name pending_contact_name in harvest.models.Property

### PR DESCRIPTION
**Pull request by farhanmasud**
Partially addresses [#116](https://github.com/LesFruitsDefendus/saskatoon-ng/issues/116)

`pending_contact_name` renamed to `pending_contact_first_name` along with all references to this field. This will keep all the existing data in `pending_contact_name` in the new `pending_contact_first_name` field.

### CODE REVIEW

**Comment by 2ynn** [comment]
@farhanmasud thanks so much! Are you working on a sqlite3 database locally? Would you feel comfortable merging the three migration files in the harvest app? If so try removing (backing up) the three migration files you created and rerun `makemigrations harvest` and `migrate` (might need to run `migrate --fake`)

**Comment by farhanmasud** [comment]
@2ynn yes, I'm currently running sqlite3 database locally. I'm merging the three new migrations into one shortly.

**Comment by farhanmasud** [comment]
@2ynn I have merged the three migration files into one.

Removing the three migration files and running the `makemigrations` command was removing the `pending_contact_name` field and adding the `pending_contact_first_name` field. Running that migration would cause the database to lose all existing data in the `pending_contact_name` field.

I have edited the new migration by taking from the previous three migration files. Running `makemigrations` (or `makemigrations harvest`) command after this edit is not detecting any new changes so I'm assuming the edit of the migration file was done correctly.

After this I have created a new testing environment of this branch by resetting the HEAD back one commit and setting up the database, loading the initial data and putting into some dummy testing data in the `Property` model. Then pulled the new code, applied the new migration and it seems to be working (keeping the old data from `pending_contact_name` field).

File: saskatoon/fixtures/saskatoon.json [hunk 1]
```diff
@@ -1 +1 @@
...euillète", "changed_by": null, "trees": [9, 14]}}, {"model": "harvest.harvest", "pk": 1, "fields": {"status": "Succeeded", "property": 3, "owner_present": true, "owner_help": true, "owner_fruit": true, "pick_leader": 2, "start_date": "2021-06-06T10:00:00Z", "end_date": "2021-06-06T10:00:00Z", "publication_date": "2021-05-27T18:43:51.899Z", "creation_date": "2021-05-26T21:42:50.701Z", "nb_required_pickers": 3, "about": "<p>Un tas de belles pommes et de noix de Grenoble au coin Mont-Royal et Buillon</p>", "changed_by": null, "trees": [9, 14], "equipment_reserved": [1]}}, {"model": "harvest.harvestyield", "pk": 1, "fields": {"harvest": 1, "tree": 9, "total_in_lb": 3.0, "recipient": 1}}, {"model": "harvest.harvestyield", "pk": 2, "fields": {"harvest": 1, "tree": 14, "total_in_lb": 6.0, "recipient": 2}}, {"model": "harvest.harvestyield", "pk": 3, "fields": {"harvest": 1, "tree": 9, "total_in_lb": 4.0, "recipient": 3}}, {"model": "harvest.harvestyield", "pk": 4, "fields": {"harvest": 1, "tree": 14, "total_in_lb": 2.0, "recipient": 1}}, {"model": "harvest.equipment", "pk": 1, "fields": {"type": 3, "description": "10ft aluminim step ladder", "owner": null, "property": 3, "shared": false}}]
```
Comment (tristanlatr) [inline saskatoon/fixtures/saskatoon.json:1]: I don't think this file needs to re-generated, right?

**Review by tristanlatr — COMMENTED** [review]
Thanks a lot @farhanmasud :)

File: saskatoon/fixtures/saskatoon.json (hunk 1 above)
Comment (farhanmasud) [reply saskatoon/fixtures/saskatoon.json:1]: I have manually updated this file to use the changed `pending_contact_first_name` field and the added `pending_contact_family_name` field. I'm don't know if this is auto generated but I have updated according to the changed models. Let me know if there's any insight about this file that I'm missing out.

Comment (tristanlatr) [reply saskatoon/fixtures/saskatoon.json:1]: @farhanmasud, I believe this file is auto-generated and the appropriate changes will be applied by the fixtures.
But @2ynn can probably speak more about this.

Comment (farhanmasud) [reply saskatoon/fixtures/saskatoon.json:1]: @tristanlatr thanks for the info. I'll make the changes according to feedback of @2ynn

**Comment by farhanmasud** [comment]
@2ynn @tristanlatr apologies, I have made some accidental changes while trying to fix the migration name. I'm fixing shortly

**Comment by farhanmasud** [comment]
@2ynn @tristanlatr really sorry for the mess up. After resolving the conflicts, the develop branch was merged and there was commits from other contributors which totally skipped my mind (working all night after a full working day, sorry). I tried to solve the issue of dependency of the newest migration (which was conflicting with the a new previous migration from another commit when migrating) and squashed from my previous commits by soft resetting the HEAD which weren't obviously my commits after the conflict resolve / merge.

I've opened a [new PR](https://github.com/LesFruitsDefendus/saskatoon-ng/pull/201) continued from here. All files should be the same as it was supposed to be here. My sincerest apologies.

**Comment by 2ynn** [comment]
no worries @farhanmasud, thanks for going through the hassle of merging the migrations, good job on that. And yes since I have now merged your other migrations for phone_fields  your changes need to build off  these.

File: saskatoon/fixtures/saskatoon.json (hunk 1 above)
Comment (2ynn) [reply saskatoon/fixtures/saskatoon.json:1]: If you want to regenerate it you'd need to first load it from a fresh database (on `develop` branch), then checkout your branch, edit the Nouveau Prorio first and last pending names from admin panel, then reexport the .json. So editing it manually for a couple changes is definitely ok. We could discuss if we really need this fixture. Note that the `saskatoon/fixtures/init` uses the individual, human readable .json files

**Pull request closed without merging**